### PR TITLE
test(core): cover openapi.ts; exclude type-only files (Phase 5)

### PR DESCRIPTION
**Pull request by hawkeyexl**
## Phase 5 of: root coverage ratchet

Two coverage improvements in one PR.

### 1. `src/core/openapi.ts` tests (23% → 96%)
Adds `test/openapi-coverage.test.js` (23 cases). `getOperation` is pure (OpenAPI definition → example/schema generation), so crafted definition fixtures drive the whole internal pipeline (`compileExample` / `getExample` / `generate{Object,Array}Example` / `getSchemas` / `getExampleParameters`): operationId lookup + guards, server resolution (explicit / `servers[0]` / throw), request + response schema extraction by code, requestBody and media-type `example`/`examples` (via `exampleKey`), parameters (path/query/header), arrays + nested objects, and `$ENV` replacement. `loadDescription` covers the required-path error and a real `$ref`-dereference success (temp-dir fixture).

Isolated coverage: **23.03% → 96.07% lines / 90.8% branches**, 100% functions. The few remaining lines are defensive guards shadowed by `getOperation`'s real args and a json-schema-faker `catch` that can't be made to throw hermetically.

### 2. Exclude type-only files from `.c8rc.json`
`dist/agents/types.js`, `dist/core/tests/wdioTypes.js`, and `dist/hints/types.js` each compile to `export {};` (pure `interface`/`type` definitions), but c8 was counting their **~314 source lines as 0%-covered** via sourcemaps — understating the real baseline with non-executable noise. They have nothing to test.

### Process note
- **No source changes** — a test file + the `.c8rc.json` exclude list.
- The combined effect (exclusions remove ~314 false-uncovered lines; openapi adds ~300 covered) should be the largest single-phase jump yet. The `coverage-thresholds.json` bump lands once the `Coverage ratchet (root)` job measures the new full-suite number.
- Hermetic: no network; temp files under `os.tmpdir()` cleaned per test.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added a comprehensive unit test suite for OpenAPI operation handling, including validation and lookup edge cases, server selection precedence, request and response example generation, parameter example behavior, and schema extraction rules.
  * Added coverage for description loading, local OpenAPI file handling, and verification that dereferenced content is processed correctly end-to-end.
* **Chores**
  * Reformatted the coverage exclusion configuration for readability without changing the excluded patterns.

### CODE REVIEW

File: test/openapi-coverage.test.js [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+          paths: {
+            "/widgets/{widgetId}": {
+              get: {
+                operationId: "getWidget",
+                parameters: [
+                  { name: "widgetId", in: "path", example: "w-123" },
+                  { name: "verbose", in: "query", example: "true" },
+                  { name: "X-Trace", in: "header", example: "trace-abc" },
+                  // A parameter whose example resolves falsy -> skipped (no push).
+                  { name: "empty", in: "query", schema: { type: "string" }, example: "" },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Reframed — the skip-branch param now has no schema/example (getExample returns null) instead of an empty-string example, so we don't encode a valid OpenAPI value as always-dropped.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/openapi-coverage.test.js [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+            },
+          },
+        }
+      );
+      const result = await getOperation(definition, "getWidget");
+      assert.equal(result.example.url, "https://api.example.com/widgets/w-123");
+      assert.equal(result.example.request.parameters.verbose, "true");
+      assert.equal(result.example.request.headers["X-Trace"], "trace-abc");
+      // The empty-value query param was dropped.
+      assert.equal(result.example.request.parameters.empty, undefined);
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Updated — assertion now checks the no-schema 'noval' param is dropped.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/openapi-coverage.test.js [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+    it("generates a param value from a parameter schema (no example present)", async function () {
+      const definition = defWith(
+        {},
+        {
+          paths: {
+            "/widgets": {
+              get: {
+                operationId: "getWidgets",
+                parameters: [
+                  { name: "limit", in: "query", schema: { type: "integer" } },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Fixed — constrained the param schema with minimum:1 so the faked value is non-falsy (a 0 would be dropped).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/openapi-coverage.test.js [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+      // json-schema-faker rather than the schema-walking fallback.
+      const definition = defWith(
+        {},
+        {
+          paths: {
+            "/widgets": {
+              get: {
+                operationId: "getWidgets",
+                parameters: [
+                  { name: "n", in: "query", required: true, type: "integer" },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Fixed — added minimum:1 to the top-level-typed param so the faked value can't be a dropped 0.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/openapi-coverage.test.js [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+  describe("requestBody example generation", function () {
+    it("generates an object body from a schema with varied property types", async function () {
+      const requestSchema = {
+        type: "object",
+        required: ["name"],
+        properties: {
+          name: { type: "string" },
+          age: { type: "integer" },
+          score: { type: "number" },
+          active: { type: "boolean" },
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Fixed — confirmed generateObjectExample also drops falsy properties (if (objectExample)), so constrained the requestBody schema: minLength:1 / minimum:1 / enum:[true]. Verified deterministic across repeated runs.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/openapi-coverage.test.js [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+      // getSchemas throws "No responses defined" when responseCode is empty and
+      // definition.responses is missing/empty. compileExample runs first, so we
+      // need it to survive: give it a usable response under an explicit code,
+      // then pass a non-matching responseCode... but getSchemas uses the same
+      // operation. Instead, exercise getSchemas' throw directly by giving the
+      // operation a single response code and requesting schemas with no code,
+      // which succeeds — so here we assert the TypeError path of an empty
+      // responses object (compileExample dereferences responses[undefined]).
+      const definition = defWith({ responses: {} });
+      await assert.rejects(() => getOperation(definition, "getWidgets"));
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/openapi-coverage.test.js]: Fixed — renamed to reflect the actual path (compileExample reads responses[undefined].headers → TypeError, before getSchemas) and assert.rejects(..., TypeError) instead of accepting any rejection.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
